pkg/ali/util/ack.ts

```diff
@@ -34,6 +34,118 @@ async function getACKOptions(
   return store.dispatch("cluster/request", { url });
 }
 
+async function getPaginatedACKOptions(store: any,
+  alibabaCredentialSecret: string,
+  regionId: string | undefined,
+  resource: string,
+  key: string,
+  extra?: object,
+  clusterId?: string) {
+
+  if (!alibabaCredentialSecret) {
+    return null;
+  }
+
+  let params: QueryParams = {
+    cloudCredentialId: alibabaCredentialSecret,
+    acceptLanguage: "en-US",
+    regionId: "us-east-1",
+  };
+
+  if (!!regionId) {
+    params.regionId = regionId;
+  }
+  if (!!clusterId) {
+    params.clusterId = clusterId;
+  }
+  if (!!extra) {
+    params = { ...params, ...extra };
+  }
+
+  let hasNext = true;
+  const out: any = [];
+  let pageNumber = 1;
+  let curSize = 0;
+  let totalItems = 0;
+  const key1 = key.split('.')[0];
+  const key2 = key.split('.')[1];
+  while (hasNext) {
+    params.pageNumber = pageNumber.toString();
+    const url = addParams(`/meta/${resource}`, params);
+    const res = await store.dispatch("cluster/request", { url });
+    curSize += res?.PageSize ? res.PageSize: 1;
+    totalItems = res?.PageSize ? res.TotalCount: 1;
```

Comment:
nit: the colon in ternary operators typically have a space before them 

```suggestion
    curSize += res?.PageSize ? res.PageSize : 1;
    totalItems = res?.PageSize ? res.TotalCount : 1;
```